jdi-light/src/main/java/com/epam/jdi/light/actions/ActionHelper.java

```diff
@@ -671,16 +672,33 @@ static boolean condition(JoinPoint jp) {
         return CONDITIONS.has(conditionName) && CONDITIONS.get(conditionName).execute(jp) || !CONDITIONS.has(conditionName) && true;
     }
 
-    public static ActionObject newInfo(ProceedingJoinPoint jp) {
+    public static ActionObject newInfo(ProceedingJoinPoint jp, String name) {
+        CHECK_MULTI_THREAD.execute();
         try {
-            return newInfo(new ActionObject(jp));
+            return newInfo(new ActionObject(jp, name));
         } catch (Throwable ex) {
             throw exception(ex, "Failed to init pjp aspect: ");
         }
     }
-    public static ActionObject newInfo(JoinPoint jp) {
+    private static long previousThread = -1;
+    public static JAction CHECK_MULTI_THREAD = () -> {
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6397461)